onnxruntime/core/optimizer/compute_optimizer/passthrough_actors.cc

```diff
@@ -0,0 +1,794 @@
+            attributes, kOnnxDomain, logger);
+  } else {
+    matmul_out_adaptor_node =
+        InsertIntermediateNodeOnDestInput(
+            graph, consumer, index,
+            0,
+            0 /* new node output index*/,
+            graph.GenerateNodeName(current_node.OpType() + "_output"),
+            "Squeeze",
+            "Squeeze node",
+            {consumer.MutableInputDefs()[index],
+             CreateUnsqueezeAxesInitializer(graph, {slice_axis})},
+            {&graph.GetOrCreateNodeArg(
+                graph.GenerateNodeArgName("squeeze_adaptor"),
+                consumer.MutableInputDefs()[index]->TypeAsProto())},
+            {}, kOnnxDomain, logger);
+  }
+
+  matmul_out_adaptor_node->SetExecutionProviderType(current_node.GetExecutionProviderType());
+
+  // Don't need set shape for Squeeze because original MatMul output is used as its output type.
+  // Set correct shape for MatMul node
+  const TensorShapeProto* matmul_out_shape = matmul_out_adaptor_node->MutableOutputDefs()[0]->Shape();
+  current_node.MutableOutputDefs()[0]->SetShape(CreateTensorShapeInsertDimAtAxis(matmul_out_shape, slice_axis, 1));
+}
+
+bool DefaultOperatorPassThroughActorBase::PostProcess(
+    Graph& graph, Node& current_node, int current_node_output_index,
+    int slice_axis, bool is_slice_scalar, bool input_has_dim_1_for_axis,
+    const ONNX_NAMESPACE::TensorShapeProto_Dimension& /*output_dim_on_axis*/,
+    const std::string& entry_node_name,
+    const std::unordered_map<int, SliceInfo>& new_gather_infos,
+    const logging::Logger& logger) {
+  LOG_DEBUG_INFO(logger, "Enter DefaultOperatorPassThroughActorBase::PostProcess for Node " + current_node.Name() +
+                             "(" + current_node.OpType() + ")");
+  if (is_slice_scalar && input_has_dim_1_for_axis) {
+    AdaptInputAndOutputForScalarSlice(graph, current_node, current_node_output_index, slice_axis,
+                                      entry_node_name, new_gather_infos, logger);
+  }
+
+  return true;
+}
+
+bool SimplePassThroughActor::PreCheck(const Graph& /*graph*/, const Node& current_node, const SliceInfo& info,
+                                      const std::vector<int>& allowed_input_indices,
+                                      const logging::Logger& logger,
+                                      std::unordered_map<int, int>& propagate_input_config,
+                                      bool& input_has_dim_1_for_axis) {
+  LOG_DEBUG_INFO(logger, "Enter SimplePassThroughActor::PreCheck for node " + current_node.Name());
+
+  Node* slice_node = info.GetNode();
+  int current_node_output_index = optimizer_utils::IndexOfNodeOutput(current_node, *slice_node->InputDefs()[0]);
+  const NodeArg* gather_data_input_arg = current_node.OutputDefs()[current_node_output_index];
+  // For each input of current_node, check whether it meets a requirements,
+  // 1). either its rank is lower than minimum_rank_to_handle.
```

Comment:
Should this be `rank is **more** than minimum_rank_to_handle`?

Reply:
Oh, the 2nd and 3rd are for rank more than minumu_rank_to_handle. 

The 1st one is for rank lower than minumu_rank_to_handle, so slice did not affect its result.

Reply:
Think twice, indeed, the comment is not accurate. refined.